### ISSUE TITLE
"Do not use BuildContexts across async gaps" in views/main_popup_menu_button.dart

### ISSUE THREAD

**Issue by davideroffo** [vandadnp/youtube-course-bloc#1]
It seems that recently flutter lint has been updated with a new linting rule that warns the developer to not use the context in async functions after an await has been performed: [Use Build Context Synchronously](https://dart-lang.github.io/linter/lints/use_build_context_synchronously.html).

<img width="720" alt="image" src="https://user-images.githubusercontent.com/6604111/210162913-6643d581-ffb3-410d-af5b-2991de656235.png">

In `step-8` we are using it in the `views/main_popup_menu_button.dart` when showing the logout and delete account dialogs for dispatching the action to the bloc:

- https://github.com/vandadnp/youtube-course-bloc/blob/main/lib/views/main_popup_menu_button.dart#L21
- https://github.com/vandadnp/youtube-course-bloc/blob/main/lib/views/main_popup_menu_button.dart#L29

Some solutions are suggesting to check whether or not the component is mounted before dispatching the event, but I was wondering if there is another solution with still using the `StatelessWidget`, without the need of converting it into a `StatefulWidget`.

**Comment by mateendev3** [comment]
yeah, please write this line before using context in the async method:

if(!mounted) return;
context.read<...>(...);
